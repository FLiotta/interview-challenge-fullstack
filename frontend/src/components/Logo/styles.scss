.logo {
  font-family: 'Montserrat';
  margin-bottom: 0px;

  &--brand-color {
    color: #7908ff;
  }
}