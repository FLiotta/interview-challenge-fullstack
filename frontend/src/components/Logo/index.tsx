// @Packages
import React from 'react';
import cn from 'classnames';

// @Own
import './styles.scss';

interface IProps {
  className?: string
  brandColor?: boolean
  tag: 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'small' | 'span'
}

const Logo: React.FC<IProps> = ({
  className,
  brandColor,
  tag
}) => {
  const getTag = () => {
    //En caso que se necesite hacer con algun otro tag para espacios mas reducidos.
    
    return React.createElement(tag, {
      className: cn("logo", className, {
        "logo--brand-color": brandColor
      })
    }, 'PiggyBank'
    );
  }

  return getTag()
}

export default Logo;