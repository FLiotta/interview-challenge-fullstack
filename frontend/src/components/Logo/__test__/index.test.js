// @Package
import { findByTestId, getByTestId, render } from '@testing-library/react';

// @Project
import Logo from '../index';

describe('Logo component', () => {
  it('Should match snapshot', () =>{
    const { container } = render(<Logo tag="p" />);

    expect(container).toMatchSnapshot();
  });

  it('Receives "className" prop', () => {
    const CLASSNAME = "prueba";
    
    const { container } = render(
      <Logo
        tag="p"
        className={CLASSNAME}
      />
    );

    expect(container.firstChild).toHaveClass(CLASSNAME);
  });

  it('Receives "tag" prop', () => {
    const TAG = "h1";

    const { container } = render(
      <Logo
        tag={TAG}
      />
    );

    const LogoWithTag = getByTestId(container, 'logo-testid');

    expect(LogoWithTag).toBeInTheDocument();
  });

  it('Receives "brandColor" prop', () => {
    const { container } = render(
      <Logo
        brandColor
        tag="h1"
      />
    );

    const LogoWithTag = getByTestId(container, 'logo-testid');

    expect(LogoWithTag).toHaveClass('logo--brand-color')
  })
});