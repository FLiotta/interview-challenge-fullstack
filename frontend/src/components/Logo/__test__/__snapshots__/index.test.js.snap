// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component Should match snapshot 1`] = `
<div>
  <p
    class="logo"
    data-testid="logo-testid"
  >
    PiggyBank
  </p>
</div>
`;
